# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1751502096 -7200
# Parent  51ac777f1797419549b21cfb111792592abede71
Branding: Make sure %APP% in URLs returns 'thunderbird' and not the string from MOZ_APP_BASENAME.

diff --git a/toolkit/components/urlformatter/URLFormatter.sys.mjs b/toolkit/components/urlformatter/URLFormatter.sys.mjs
--- a/toolkit/components/urlformatter/URLFormatter.sys.mjs
+++ b/toolkit/components/urlformatter/URLFormatter.sys.mjs
@@ -105,9 +105,13 @@ nsURLFormatterService.prototype = {
     PLATFORMBUILDID() {
       return Services.appinfo.platformBuildID;
     },
     APP() {
-      return Services.appinfo.name.toLowerCase().replace(/ /, "");
+      // Services.appinfo.name delivers "Betterbird" via MOZ_APP_BASENAME,
+      // which is put into `mAppData->name` in toolkit/xre/nsAppRunner.cpp
+      // by parsing application.ini in XRE_ParseAppData().
+      // We want %APP% in URLs to be "thunderbird".
+      return "thunderbird"; // Services.appinfo.name.toLowerCase().replace(/ /, "");
     },
     OS() {
       return Services.appinfo.OS;
     },
